### PR TITLE
feat: use schema's title when available in `{one,any,all}Of` applicators

### PR DESCRIPTION
**Pull request by KtorZ**
**Description**

Applicators can be a little hard to navigate, especially on deeply nested schema. While the `uid` provides a nice middle-ground solution, it doesn't actually work when schemas are defined outside of the main specification and more importantly, it isn't necessarily meant to be human-readable. The `title` property however is meant just for that.

Changes proposed in this pull request:

- This PR does just that, it adds the `title` next to branches enumeration in `{one,any,all}Of` applicators. When not available, it defaults to the current behavior.

image

**Related issue(s)**

N/A

### CODE REVIEW

**Review by fmvilas — APPROVED** [review]
LGTM 👍

**Comment by fmvilas** [comment]
@KtorZ would you mind merging the `next` branch into your branch? I'd do it myself but Github doesn't let me.

**Comment by KtorZ** [comment]
I'll try to get this done in the upcoming days!

**Comment by KtorZ** [comment]
@fmvilas done 👌

**Comment by derberg** [comment]
`next` will soon be deleted. It is already merged into `master` and 1.0 is now available.

 no more long living release branches!

sorry for any trouble with your PR 😞 but we really had to finally release this v1 and get rid of release branch

maybe changing target branch from `next` to `master` will not cause to much trouble and conflicts. Can you try 🙏🏼

**Comment by KtorZ** [comment]
No worries. Just switching the branch didn't do it (more than 100k lines changed for some reason). So I just pulled the latest master and cherry-picked the commit onto it. Seems like it's good. 

Let me know 🙏

File: library/src/components/Schema.tsx
Comment (derberg) [inline library/src/components/Schema.tsx:313]: wouldn't it be better if `applicatorSchemaName` just accepts 2 arguments?

`SchemaHelpers.applicatorSchemaName(idx, s)` or `SchemaHelpers.applicatorSchemaName(idx, s.title())` as testing will be easier

**Review by derberg — COMMENTED** [review]
Thank sfor your patience. I left one small comment

File: library/src/components/Schema.tsx
Comment (KtorZ) [reply library/src/components/Schema.tsx:313]: What would be better/easier exactly 🤔? I don't see how we make this work without either: 

- duplicating the pattern-match logic on the schema applicator kind inside the `applicatorSchemaName` function.

- or, duplicating the title creation logic on the index number at every call site. 

Do you have a more concrete example of what you have in mind?

Comment (derberg) [reply library/src/components/Schema.tsx:313]: oh crap, forget it, sorry for monday blindness... I seriously did not notice the other usage of `applicatorSchemaName` usage 🤦🏼

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
@KtorZ please fix linter issues and also update with latest master

**Comment by derberg** [comment]
there is just one tiny fix

```
ERROR: (triple-equals) /home/runner/work/asyncapi-react/asyncapi-react/library/src/helpers/schema.ts[110, 27]: != should be !==
```

I cannot commit it directly for you as you open a PR not from personal but for organization fork

**Comment by KtorZ** [comment]
I'll take care of it tomorrow 🫡

**Comment by derberg** [comment]
@KtorZ kind ping 😉

**Comment by KtorZ** [comment]
My bad. Tomorrow ended up being way further in the future than I anticipated. Linted + rebased on top of `master`

**Comment by derberg** [comment]
SonarCloud blocked merge because of new rule related to accessibility. Issue mitigated by creating https://github.com/asyncapi/asyncapi-react/issues/827

**Comment by derberg** [comment]
@KtorZ all looks good but sorry, cannot merge as I cannot update your branch on my own as it is an org-fork, you have to do it on your own again, sorry

**Comment by derberg** [comment]
@KtorZ whenever you update your fork with latest master, just ping me directly here or in slack, so i will quickly merge

**Comment by KtorZ** [comment]
@derberg 🙏

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
@KtorZ finally we did it 😄

**Comment by KtorZ** [comment]
Easy enough ... 😶

**Comment by derberg** [comment]
yeah, organization forks are pain, at least for upstream maintainers 😃

File: library/src/helpers/schema.ts
Comment (kennethaasan) [inline library/src/helpers/schema.ts:110]: The title can also be `undefined` it seems. This introduced `Consists of undefined:` for us. So this check needs to be improved.

**Pull request merged**
